apps/backend/src/libs/plugins/auth-plugin/auth-plugin.ts

```diff
@@ -0,0 +1,38 @@
+import { type FastifyPluginCallback } from "fastify";
+
+import { HTTPCode } from "~/libs/modules/http/libs/enums/enums.js";
+import { ServerHooks } from "~/libs/plugins/libs/enums/enums.js";
+
+type PluginOptions = {
+	excludedRoutePrefixes?: string[];
+	tokenVerifier: (token: string) => Promise<unknown>;
+};
+
+const authPlugin: FastifyPluginCallback<PluginOptions> = (
+	app,
+	{ excludedRoutePrefixes = [], tokenVerifier },
+) => {
+	app.addHook(ServerHooks.PRE_HANDLER, async (request, reply) => {
+		for (const routePrefix of excludedRoutePrefixes) {
+			if (request.url.startsWith(routePrefix)) {
+				return;
+			}
+		}
+
+		const token = request.headers.authorization;
+
+		if (!token) {
+			return await reply
+				.code(HTTPCode.UNAUTHORIZED)
+				.send({ message: "Authorization header required for this route" });
+		}
```

Comment:
throw AuthError instead

Reply:
Maybe let's move the error message to a separate enum?

Reply:
Maybe let's move the error message to a separate enum?

Reply:
Can't find `AuthError`. Do I need to create it or just use the `HTTPError` from "shared" lib?

Reply:
Check #37, AuthError was added there. You should add it as well